tests/jballerina-unit-test/src/test/resources/test-src/types/typereftype/type_reference.bal

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+type Nil ();
+type Integer int;
+type Decimal decimal;
+type Strings string;
+type IntArray int[2];
+type FloatBooleeanTuple [float, boolean];
+type Record record {|
+    FloatBooleeanTuple a;
+    Nil b;
+|};
+type FunctionType1 function (Integer i) returns Decimal;
+type FunctionType2 function (FloatBooleeanTuple i) returns Record;
+
+type Object1 object {
+    public int a;
+    *Object2;
+};
+
+type Object2 object {
+    public int b;
+    *Object3;
+};
+
+type Object3 object {
+    public int c;
+};
+
+
+type One 1;
+type ImmutableIntArray int[] & readonly;
+type ImmutableIntOrStringArray string[]|ImmutableIntArray;
```

Comment:
Is the name correct here? IMO it reads like `(int|string)[] & readonly` rather than `(int[] & readonly)|string[]`.

Reply:
`ImmutableIntArrayOrStringArray`, is this okay?